google/cloud/spanner/client_test.cc

```diff
@@ -1026,59 +1034,86 @@ TEST(ClientTest, ProfileQueryWithOptionsSuccess) {
 }
 
 TEST(ClientTest, QueryOptionsOverlayPrecedence) {
-  struct Levels {
-    absl::optional<std::string> env;
-    absl::optional<std::string> client;
-    absl::optional<std::string> function;
-    absl::optional<std::string> expected;
-  };
-  std::vector<Levels> levels = {
-      {{}, {}, {}, {}},
-      {"env", {}, {}, "env"},
-      {{}, "client", {}, "client"},
-      {{}, {}, "function", "function"},
-      {"env", "client", {}, "client"},
-      {"env", {}, "function", "function"},
-      {{}, "client", "function", "function"},
-      {"env", "client", "function", "function"},
-  };
+  constexpr auto OverlayQueryOptions =  // NOLINT(readability-identifier-naming)
```

Comment:
I am sure you understand the tradeoffs between testing the function that computes the query options vs. verifying the correct query options are sent to the `SpannerConnection`.  This note is just a nudge to consider these tradeoffs one last time before you pull the trigger.  I am good either way.

Reply:
There is a separate test, `ConnectionImplTest.QueryOptions`, for verifying how a final `QueryOptions` value is propagated into the protos.  It is deficient in the `request_priority` department however, so I'll fix that in another PR.